### PR TITLE
DO NOT MERGE: Watcher hot shit

### PR DESCRIPTION
**Pull request by remissio**

### CODE REVIEW

File: iac-manager/iacmanager.go [hunk 1]
```diff
@@ -136,7 +141,9 @@ func addRemote() error {
 	}
 	err = gitmanager.CheckoutBranch(utils.CONFIG.Kubernetes.GitVaultDataPath, utils.CONFIG.Iac.RepoBranch)
 	if err != nil {
-		iaclogger.Errorf("Error setting up branch: %s", err.Error())
+		if err.Error() != "remote repository is empty" {
```
Comment (beneiltis) [inline iac-manager/iacmanager.go]: hier muss gegen den error gecheckt werden nicht gegen string

File: iac-manager/iacmanager.go [hunk 2]
```diff
@@ -530,6 +537,11 @@ func SyncChanges() error {
 						SetPulseDiagramData(pulse)
 					}
 				}
+				// skipp this error
+				if err.Error() == "remote repository is empty" {
```
Comment (beneiltis) [inline iac-manager/iacmanager.go]: gleicher scheisss

File: kubernetes/watcher.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+	obj, err := GetObjectFromFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	k8sObject, err := provider.DynamicClient.Resource(schema.GroupVersionResource{
+		Group:    obj.GroupVersionKind().Group,
+		Version:  obj.GroupVersionKind().Version,
+		Resource: GetResourceNameForUnstructured(obj),
+	}).Namespace(obj.GetNamespace()).Get(context.TODO(), obj.GetName(), metav1.GetOptions{})
```
Comment (beneiltis) [inline kubernetes/watcher.go]: unterscheidung namespaced non namespaced methode

File: kubernetes/watcher.go [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+	if err != nil {
+		fmt.Println(strings.ToLower(obj.GetKind())+"s", obj.GetNamespace(), obj.GetName(), obj.GroupVersionKind().Group, obj.GroupVersionKind().Version)
```
Comment (beneiltis) [inline kubernetes/watcher.go]: raus die fmts.

File: utils/config.go [hunk 5]
```diff
@@ -190,46 +233,41 @@ func InitConfigYaml(showDebug bool, customConfigName string, stage string) {
 	}
 
 	// SET DEFAULTS if missing
-	pwd, _ := os.Getwd()
+	tempDirPath, err := os.MkdirTemp("", "mo_*")
+	if err != nil {
+		log.Fatalf("Failed to create temp dir: %v", err)
+	}
+	log.Infof("TemoDir created: %s", tempDirPath)
```
Comment (beneiltis) [inline utils/config.go]: typo

File: iac-manager/iacmanager.go (hunk 1 above)
Comment (remissio) [reply iac-manager/iacmanager.go]: i'll handle it

just for reference; thats the source:

https://github.com/go-git/go-git/blob/790b7402e72946d94a057497211ba9c770071703/plumbing/transport/common.go#L34

File: iac-manager/iacmanager.go (hunk 2 above)
Comment (remissio) [reply iac-manager/iacmanager.go]: i'll handle it

just for reference; thats the source:

https://github.com/go-git/go-git/blob/790b7402e72946d94a057497211ba9c770071703/plumbing/transport/common.go#L34

**Pull request merged**